power_grid_model_c/power_grid_model/include/power_grid_model/auxiliary/dataset.hpp

```diff
@@ -104,6 +220,8 @@ template <dataset_type_tag dataset_type_> class Dataset {
     Buffer const& get_buffer(std::string_view component) const { return get_buffer(find_component(component, true)); }
     Buffer const& get_buffer(Idx i) const { return buffers_[i]; }
 
+    bool is_columnar(Data* data) const { return data == nullptr; }
```

Comment:
this usage looks weird... maybe it's better to make this the private implementation and only `public`ly expose `bool is_columnar(std::string_view component)` and `bool is_columnar(Idx i)` and optionally `bool is_columnar(Buffer const& buffer)`

or (unlikely) you can add this to the `ComponentInfo` of a certain component

Reply:
I realize how the original implementation of `is_columnar(Data* data)` is very weird. I think that leaving the three implementations that you suggest and remove the original should be better.